lib/pwwka/queue_resque_job_handler.rb

```diff
@@ -4,16 +4,36 @@
 module Pwwka
   # A handler that simply queues the payload into a Resque job.  This is useful
   # if the code that should respond to a message needs to be managed by Resque, e.g.
-  # for the purposes of retry or better failure management.
+  # for the purposes of retry or better failure management.  You can ask for the routing key and properties
+  # by setting `PWWKA_QUEUE_EXTENDED_INFO` to `true` in your environment.
   #
   # You should be able to use this directly from your handler configuration, e.g. for a Heroku-style `Procfile`:
   #
   #     my_handler: rake message_handler:receive HANDLER_KLASS=Pwwka::QueueResqueJobHandler JOB_KLASS=MyResqueJob QUEUE_NAME=my_queue ROUTING_KEY="my.key.completed"
+  #     my_handler_that_wants_more_info: rake message_handler:receive HANDLER_KLASS=Pwwka::QueueResqueJobHandler JOB_KLASS=MyOthgerResqueJob PWWKA_QUEUE_EXTENDED_INFO=true QUEUE_NAME=my_queue ROUTING_KEY="my.key.#"
   #
   # Note that this will not check the routing key, so you should be sure to specify the most precise ROUTING_KEY you can for handling the message.
   class QueueResqueJobHandler
     def self.handle!(delivery_info,properties,payload)
-      Resque.enqueue(ENV["JOB_KLASS"].constantize, payload)
+      job_klass = ENV["JOB_KLASS"].constantize
+      args = [
+        job_klass,
+        payload
+      ]
+      if ENV["PWWKA_QUEUE_EXTENDED_INFO"] == 'true' || job_klass_can_handle_args?(job_klass)
+        args << delivery_info.routing_key
+        args << properties.to_hash
+      end
+      Resque.enqueue(*args)
+    end
+
+  private
+
+  def self.job_klass_can_handle_args?(job_klass)
+      method = job_klass.method(:perform)
+      return false if method.nil?
+      method.arity == 3
```

Comment:
@danshultz what do you think?  I know there are other compatible signatures that aren't arity 3, but this felt like a clear, conservative, obvious signal that the job can handle these args.

Reply:
This looks good - I would also check for an arity of `-2` as I would expect those updating their handlers do the following. However one could create new handlers but I am not sure that's most efficient.

```
self.perform(payload, key=nil, properties=nil)
end
```

Reply:
I would expect this because it's less moving parts to ensure old messages are handled and deployments do not have to be thought about. But maybe upgrading old handlers is not really in the use case.